src/core/reference/src/utils/phillox_converter.cpp

```diff
@@ -0,0 +1,513 @@
+template <typename T, typename V>
+PytorchAccumulatorType<T> uint32_to_T_type_float(V x) {
+    const auto mask = static_cast<V>((uint64_t(1) << std::numeric_limits<T>::digits) - 1);
+    const auto divisor = static_cast<PytorchAccumulatorType<T>>(1) / (uint64_t(1) << std::numeric_limits<T>::digits);
+    PytorchAccumulatorType<T> ret = (x & mask) * divisor;
+    return ret;
+}
+
+float uint32_to_bfloat16_pytorch(uint32_t x) {
+    const auto mask = static_cast<uint32_t>((1UL << 8) - 1);
+    const auto divisor = static_cast<PytorchAccumulatorType<ov::bfloat16>>(1) / (1UL << 8);
+    PytorchAccumulatorType<ov::bfloat16> ret = (x & mask) * divisor;
+    return ret;
+}
+
+// Converts uint32 values to destination type and normalizes to required range.
+template <typename T>
+void convert_to_output_type(const std::vector<uint32_t>& res,
+                            size_t uints_generated_per_exec,
+                            size_t output_type_generated_values,
+                            const element::Type& elem_type,
+                            const char* min_val,
+                            const char* max_val,
+                            char* out,
+                            size_t k,
+                            size_t elem_count,
+                            T (*convert_single_input)(uint32_t) = nullptr,
+                            T (*convert_two_inputs)(uint32_t, uint32_t, T, T) = nullptr,
+                            T (*mod_func)(uint32_t, T, T) = nullptr) {
+    // Get min and max values
+    T mn[1];
+    T mx[1];
+    memcpy(mn, min_val, elem_type.size());
+    memcpy(mx, max_val, elem_type.size());
+
+    std::vector<T> res_out_type(uints_generated_per_exec);
+    // Either 2 or 4 (2 for 64 bit values, 4 otherwise)
+    if (output_type_generated_values == 2) {
+        // Each element of resulting sequence is formed using two uint32 values,
+        // therefore only half of the res_out_type array is used.
+        for (size_t i = 0; i < uints_generated_per_exec / 2; ++i) {
+            res_out_type[i] = convert_two_inputs(res[2 * i], res[2 * i + 1], mn[0], mx[0]);
+        }
+    } else {
+        // Each element of resulting sequence is formed using single uint32 value
+        std::transform(res.data(),
+                       res.data() + uints_generated_per_exec,
+                       res_out_type.data(),
+                       [&mn, &mx, &convert_single_input, &mod_func](uint32_t elem) {
+                           if (convert_single_input != nullptr) {
+                               return convert_single_input(elem) * (mx[0] - mn[0]) + mn[0];
+                           } else {
+                               return mod_func(elem, mn[0], mx[0]);
+                           }
+                       });
+    }
+
+    memcpy(out + k * elem_type.size(),
+           res_out_type.data(),
+           std::min(uints_generated_per_exec, elem_count - k) * elem_type.size());
```

Comment:
Shouldn't it compare `output_type_generated_values` instead?
```suggestion
    memcpy(out + k * elem_type.size(),
           res_out_type.data(),
           std::min(output_type_generated_values, elem_count - k) * elem_type.size());
```

Reply:
Yes, replaced